modules/user_accounts/ajax/rejectUser.php

```diff
@@ -1,122 +1,63 @@
+ * user sending the request has admin permissions and the account
+ * can be removed (no password hash => has never had an activity on
+ * Loris, and is pending).
+ *
+ * PHP Version 7
  *
  * @category Loris
  * @package  User_Accounts
  * @author   Leo T. <lthomas.mcin@gmail.com>
+ *           Zaliqa Rosli <zaliqa.rosli@mcin.ca>
  * @license  Loris license
  * @link     https://github.com/aces/Loris
  */
 
-    define('NO_IDENTIFIER_SUPPLIED', 1);
-    define('USER_NOT_FOUND', 2);
-    define('INCORRECT_PERMISSION', 3);
-    define('ACCOUNT_ACTIVE', 4);
-
+namespace LORIS\user_accounts;
 
-if (!isset($_POST['identifier'])) {
-    throw new LorisException("No identifier supplied", NO_IDENTIFIER_SUPPLIED);
-    exit(1);
-} else {
-    $identifier = $_POST["identifier"];
-    reject_user($identifier);
-}
-/**
-* Checks that logged in user has user_accounts permissions, which is
-* also included in admin permissions
-*
-* @return boolean true if user has admin permission or user accounts false otherwise
-*/
-function _hasPerm()
-{
-    $user =& User::singleton();
-    if (isset($user)) {
-        if (!$user->hasPermission('user_accounts')) {
-            return false;
-        }
-        return true;
-    }
-    return false;
+// Make sure the permission is checked first to avoid malicious userID scans
+if (!(\User::singleton())->hasPermission('user_accounts')) {
+    header("HTTP/1.1 403 Forbidden");
+    header("Content-Type: text/plain");
+    exit(
+        "You do not have the correct permissions for this 
+         operation (need admin or user accounts)"
+    );
 }
 
-    /**
-    * Query database on UserID to check that password has doesn't
-    * exist and account is still pending approval, to ensure the
-    * users with account activity can't be rejected by setting
-    * their status to pending.
-    *
-    * @param string $identifier of account to verify
-    *
-    * @return boolean true if Password_hash is null and
-    * Pending_approval is Yes, false otherwsie
-    *
-    * @throws Loris exception if the userID isn't found in the database.
-    */
-function _canRejectAccount($identifier)
-{
-    $DB = Database::singleton();
-    try{
-        $fields =  $DB->pselect(
-            "SELECT Password_hash, Pending_approval
-                       FROM users
-                       WHERE UserID=:id",
-            array("id" => $identifier)
-        );
-        if (is_null($fields[0]['Password_hash'])
-            && $fields[0]['Pending_approval']=="Y"
-        ) {
-            return true;
-        }
-        return false;
-    } catch (DatabaseException $e) {
-        throw new LorisException("User not found in database", USER_NOT_FOUND);
-        exit(1);
-    }
+if (!isset($_POST['identifier'])) {
+    header("HTTP/1.1 400 Bad Request");
+    header("Content-Type: text/plain");
+    exit(
+        "No identifier supplied"
+    );
 }
 
+$rejectee = \User::factory($_POST['identifier']);
+$username = $rejectee->getUsername();
```

Comment:
This will probably work, but it should probably also check to make sure $rejectee is not an instanceof `\LORIS\AnonymousUser` in case `getUsername()` is ever implemented for the AnonymousUser class to return something other than the empty string (which is just coincidental right now.)

Reply:
@xlecours this is for you haha

Reply:
Good point.. but this looks like a significant change in design that it should probably be going to major anyways to get thoroughly tested before a release..